File: java/adapter/WeatherServiceAdaptee.java
package adapter;

public class WeatherServiceAdaptee implements WeatherServiceAdapter {

    private final LegacyWeatherService legacyWeatherService;

    public WeatherServiceAdaptee(LegacyWeatherService legacyWeatherService) {
        this.legacyWeatherService = legacyWeatherService;
    }

    public TemperatureDate getTemperature(double longitude, double latitude) {
        String temperatureDataInXML = legacyWeatherService.getTemperature(getCityOf(longitude, latitude),
                getCountryOf(longitude, latitude));
        return convertXMLDataToJson(temperatureDataInXML);
    }

    private TemperatureDate convertXMLDataToJson(String xmlData) {
        System.out.println("Converting...");
        return new TemperatureDate("Converted Data from XML into JSON");
    }

    private String getCityOf(double longitude, double latitude) {
        System.out.println("Extracting city of longitude: " + longitude + " and latitude: " + latitude);
        return "City";
    }

    private String getCountryOf(double longitude, double latitude) {
        System.out.println("Extracting country of longitude: " + longitude + " and latitude: " + latitude);
        return "Country";
    }
}

File: typescript/adapter/weather-service-adaptee.ts
import { LegacyWeatherService } from "./legacy-weather-service";
import { TemperatureData } from "./temperature-data";
import { WeatherServiceAdapter } from "./weather-service-adapter";

export class WeatherServiceAdaptee implements WeatherServiceAdapter {
  private readonly legacyWeatherService: LegacyWeatherService;

  constructor(legacyWeatherService: LegacyWeatherService) {
    this.legacyWeatherService = legacyWeatherService;
  }

  getTemperature(longitude: number, latitude: number): TemperatureData {
    const temperatureDataInXML = this.legacyWeatherService.getTemperature(
      this.getCityOf(longitude, latitude),
      this.getCountryOf(longitude, latitude)
    );
    return this.convertXMLDataToJson(temperatureDataInXML);
  }

  private convertXMLDataToJson(xmlData: string): TemperatureData {
    console.log("Converting...");
    return new TemperatureData("Converted Data from XML into JSON");
  }

  private getCityOf(longitude: number, latitude: number): string {
    console.log(
      `Extracting city of longitude: ${longitude} and latitude: ${latitude}`
    );
    return "City";
  }

  private getCountryOf(longitude: number, latitude: number): string {
    console.log(
      `Extracting country of longitude: ${longitude} and latitude: ${latitude}`
    );
    return "Country";
  }
}
